pkg/comp-functions/functions/common/resources.go

```diff
@@ -3,43 +3,111 @@ package common
 import (
 	vshnv1 "github.com/vshn/appcat/v4/apis/vshn/v1"
 	"github.com/vshn/appcat/v4/pkg/common/utils"
+	"k8s.io/apimachinery/pkg/api/resource"
 )
 
 type Resources struct {
-	ReqMem string
-	ReqCPU string
-	Mem    string
-	CPU    string
-	Disk   string
+	ReqMem resource.Quantity
+	ReqCPU resource.Quantity
+	Mem    resource.Quantity
+	CPU    resource.Quantity
+	Disk   resource.Quantity
 }
 
-func GetResources(size *vshnv1.VSHNSizeSpec, r utils.Resources) Resources {
-	reqMem := size.Requests.Memory
-	reqCPU := size.Requests.CPU
-	mem := size.Memory
-	cpu := size.CPU
-	disk := size.Disk
+func GetResources(size *vshnv1.VSHNSizeSpec, plan utils.Resources) (Resources, error) {
+	reqMem := resource.Quantity{}
+	reqCPU := resource.Quantity{}
+	mem := resource.Quantity{}
+	cpu := resource.Quantity{}
+	disk := plan.Disk
 
-	if reqMem == "" {
-		reqMem = r.MemoryRequests.String()
+	var err error
+
+	if size.Requests.Memory != "" {
+		reqMem, err = resource.ParseQuantity(size.Requests.Memory)
+		if err != nil {
+			return Resources{}, err
```

Comment:
Consider not returning any error here but just continue, you would have an empty value which will be discarded later on in the logic. I fear we might create more problems by returning here an error. Applies to the rest of the variables as well.

Reply:
Might it make sense, to collect the errors and return the errors to the main routine and if there are errors, just print them out?

Reply:
Yes that is a good compromise. We should not see any errors of this type here as I expect the validation to happen much earlier in the logic. So yeah let's gather all the errors and then return as a warning, might make sense that way.